test/tcp/TcpClient.cpp

```diff
@@ -27,6 +28,46 @@ class Callback : public AsyncSocket::ConnectCallback {
     std::cout << "TODO error" << ex.what() << " " << ex.getType() << "\n";
   }
 };
+
+class FollyKeepaliveTimer : public KeepaliveTimer {
+ public:
+  explicit FollyKeepaliveTimer(folly::EventBase& eventBase)
+      : eventBase_(eventBase) {
+    running_ = std::make_shared<bool>(false);
+  }
+
+  std::chrono::milliseconds keepaliveTime() override {
+    return std::chrono::milliseconds(5000);
+  };
+
+  void schedule() {
```

Comment:
those asserts would involve remembering current thread at least.. I am not sure it that's what you want... 
Our implementation has such assumptions on a lot of places. I would be fine with just a comment, but am open to anything.